tsl/src/continuous_aggs/common.c

```diff
@@ -587,16 +587,17 @@ get_bucket_width(CAggTimebucketInfo bucket_info)
 			 * result that year is not multiple of day or month, which is perceived
 			 * as a bug. For that reason, we treat all months as 30 days regardless of year
 			 */
-			if (bucket_info.interval->month && !bucket_info.interval->day &&
-				!bucket_info.interval->time)
+			Interval *interval = palloc(sizeof(Interval));
+			memcpy(interval, bucket_info.interval, sizeof(Interval));
```

Comment:
Would be good to add some comments here why we're copying the value of `bucket_info.interval` instead of use it